client/src/main/java/io/fabric8/docker/client/impl/ExecNamedOperationImpl.java

```diff
@@ -58,30 +60,33 @@ public Boolean resize(int h, int w) {
 
     @Override
     public Boolean start() {
+        ExecStartCheck config = new ExecStartCheckBuilder().withDetach(true).withTty(false).build();
+        return start(config);
+    }
+
+    public Boolean start(ExecStartCheck config) {
         try {
             StringBuilder sb = new StringBuilder();
             sb.append(getResourceUrl());
-            RequestBody body = RequestBody.create(MEDIA_TYPE_TEXT, EMPTY);
+            RequestBody body = RequestBody.create(MEDIA_TYPE_JSON, JSON_MAPPER.writeValueAsString(config));
             Request.Builder requestBuilder = new Request.Builder().post(body).url(URLUtils.join(getResourceUrl().toString(), START_OPERATION));
-            handleResponse(requestBuilder, 204);
+            handleResponse(requestBuilder, 200);
             return true;
         } catch (Exception e) {
             throw DockerClientException.launderThrowable(e);
         }
     }
 
+    //size not used in inspec exec API
     @Override
-    public ContainerInspect inspect() {
-        return inspect(false);
+    public ContainerInspect inspect(Boolean withSize) {
+        return inspect();
     }
 
     @Override
-    public ContainerInspect inspect(Boolean withSize) {
-        StringBuilder sb = new StringBuilder();
+    public ContainerInspect inspect() {
         try {
-            sb.append(getResourceUrl());
-            sb.append(Q).append(SIZE).append(EQUALS).append(withSize);
```

Comment:
if we are to ignore withSize (e.g. is not applicable to this operation) we should also remove ``inspect(Boolean withSize)``.